tests/test_integration.py

```diff
@@ -532,3 +535,153 @@ def create_layout():
+        def update_output(value):
+            return value
+
+        self.startServer(app)
+
+        input1 = self.wait_for_element_by_id('input')
+        input1.clear()
+
+        input1.send_keys('fire request hooks')
+
+        output1 = self.wait_for_element_by_id('output-1')
+        output_pre = self.wait_for_element_by_id('output-pre')
+        output_post = self.wait_for_element_by_id('output-post')
+
+        self.assertEqual('fire request hooks', output1.text)
+        self.assertEqual('request_pre changed this text!', output_pre.text)
+        self.assertEqual('request_post changed this text!', output_post.text)
+        self.percy_snapshot(name='request-hooks')
+
+    def test_with_custom_renderer_interpolated(self):
+        app = dash.Dash(__name__)
+
+        app.layout = html.Div([
+            dcc.Input(
+                id='input',
+                value='initial value'
+            ),
+            html.Div(
+                html.Div([
+                    html.Div(id='output-1'),
+                    html.Div(id='output-pre'),
+                    html.Div(id='output-post')
+                ])
+            )
+        ])
+
+        @app.callback(Output('output-1', 'children'), [Input('input', 'value')])
+        def update_output(value):
+            return value
+
+        scripts = app._generate_scripts_html()
+        css = app._generate_css_dist_html()
+        config = app._generate_config_html()
+        metas = app._generate_meta_html()
+        title = "test custom renderer on interpolated_index"
+        favicon = ''
+        _app_entry = '''
+        <div id="react-entry-point">
+            <div class="_dash-loading">
+                Loading...
+            </div>
+        </div>
+        '''
+        renderer = '''
+        <script id="_dash-renderer" type"application/javascript">
+            console.log('firing up a custom renderer!')
+            const renderer = new DashRenderer({
+                request_pre: () => {
+                    var output = document.getElementById('output-pre')
+                    if(output) {
+                        output.innerHTML = 'request_pre changed this text!';
+                    }
+                },
+                request_post: () => {
+                    var output = document.getElementById('output-post')
+                    if(output) {
+                        output.innerHTML = 'request_post changed this text!';
+                    }
+                }
+            })
+        </script>
+        '''
+
+        index = app.interpolate_index(
+            metas=metas, title=title, css=css, config=config,
+            scripts=scripts, app_entry=_app_entry, favicon=favicon,
+            renderer=renderer)
+
+        self.startServer(app)
+
+        input1 = self.wait_for_element_by_id('input')
+        input1.clear()
+
+        input1.send_keys('fire request hooks')
+
+        output1 = self.wait_for_element_by_id('output-1')
```

Comment:
these items should already be on the page at this point, so I don't think that there will be any waiting that will happen. However, `send_keys` will fire off several async callback functions, so you will likely need to _wait_ for `output1.text` text to equal "fire request hooks". In some other dash projects, we had a utility `wait_for_text_to_equal` that would be good here, or there might be something native in selenium